packages/test/local-server-stress-tests/src/test/localServerStressOrderSequentially.spec.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { done, isOperationType, takeAsync } from "@fluid-private/stochastic-test-utils";
+
+import { makeGenerator, reducer, saveFailures, type StressOperations } from "../baseModel.js";
+import {
+	convertToRealHandles,
+	covertLocalServerStateToDdsState,
+	DDSModelOpGenerator,
+	loadAllHandles,
+	validateConsistencyOfAllDDS,
+	type DDSModelOp,
+	type OrderSequentially,
+} from "../ddsOperations.js";
+import {
+	createLocalServerStressSuite,
+	LocalServerStressModel,
+	type LocalServerStressState,
+} from "../localServerStressHarness.js";
+
+const orderSequentiallyReducer = async (
+	state: LocalServerStressState,
+	op: OrderSequentially,
+) => {
+	const { baseModel, taggedHandles } = await loadAllHandles(state);
+	const ddsState = await covertLocalServerStateToDdsState(state);
+	const rollbackError = new Error("rollback");
+	try {
+		state.datastore.orderSequentially(() => {
+			for (const o of op.operations) {
+				baseModel.reducer(ddsState, convertToRealHandles(o, taggedHandles));
+				if (op.rollback) {
+					throw rollbackError;
+				}
+			}
+		});
+	} catch (error) {
+		if (error !== rollbackError) {
+			throw error;
+		}
+	}
+};
+
+describe("Local Server Stress with rollback", () => {
+	const model: LocalServerStressModel<StressOperations | OrderSequentially> = {
+		workloadName: "rollback",
+		generatorFactory: () =>
+			takeAsync(
+				100,
+				makeGenerator<OrderSequentially>([
+					[
+						async (state) => {
+							const operations: DDSModelOp[] = [];
+							/**
+							 * unfortunately we can't generate more than a single op here, as each op is generated off
+							 * the current state, and if we generate multiple ops it can result in earlier ops invaliding
+							 * the constrains necessary for later ops. for example, an earlier op might delete a sub-directory
+							 * which a later op sets a key in, but the state and generator don't know that will happen.
+							 */
+							const op = await DDSModelOpGenerator(state);
+							if (op !== done) {
+								operations.push(op);
+							}
+							return {
+								type: "orderSequentially",
+								operations,
+								rollback: true,
```

Comment:
This makes me wonder why not just update the DDS model generators to have a boolean rollback which is sometimes `true`.

I guess... this is basically doing that, by wrapping it rather than updating it.  But then why bother with this boolean property?  Maybe just call it `RolledBackDDSModelOp` rather than `OrderSequentially`?

Hopefully I'm following all this correctly

Reply:
i didn't have it always rollback to start, but then i decided it was more useful to test rollback deeply, as order sequentially without rollback isn't very interesting. so this is built suck that we could test order sequentially without rollback if we wanted to.

i don't think it make sense on dds, as dds can't rollback, only order sequentially can. ideally, we'd be ablet o generate multiple ops inside an order sequentially, but it is a bit tricky, but i do have some ideas how to accomplish it, and then it will make even less sense to do it per op, as we are rolling back the whole batch

Reply:
Maybe a good motivation for moving to a more traditional generator that can inspect state and generate ops one-at-a-time as they are used.

Reply:
i don't think that would help. the problem is really about asynchrony which is hard to resolve regardless of the shape.